msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -712,6 +712,40 @@ public void acquireToken(@NonNull final Activity activity,
         );
     }
 
+    /**
+     * Acquire token interactively, will pop-up webUI. Interactive flow will skip the cache lookup.
+     * Default value for {@link UiBehavior} is {@link UiBehavior#SELECT_ACCOUNT}.
+     *
+     * @param activity    Non-null {@link Activity} that is used as the parent activity for launching the {@link AuthenticationActivity}.
+     *      *                 All apps doing an interactive request are required to call the
+     *      *                 {@link PublicClientApplication#handleInteractiveRequestRedirect(int, int, Intent)} within the calling
+     *      *                 activity {@link Activity#onActivityResult(int, int, Intent)}.
+     * @param resource    The non-null string of resource identifier.
+     * @param callback    The {@link AuthenticationCallback} to receive the result back.
+     *      *                 1) If user cancels the flow by pressing the device back button, the result will be sent
+     *      *                 back via {@link AuthenticationCallback#onCancel()}.
+     *      *                 2) If the sdk successfully receives the token back, result will be sent back via
+     *      *                 {@link AuthenticationCallback#onSuccess(IAuthenticationResult)}
+     *      *                 3) All the other errors will be sent back via
+     *      *                 {@link AuthenticationCallback#onError(MsalException)}.
+     */
+    public void acquireToken(@NonNull final Activity activity,
+                             @NonNull final String resource,
+                             @NonNull final AuthenticationCallback callback) {
+        acquireToken(
+                activity,
+                mapResourceToDefaultScope(resource),
```

Comment:
I also just realized we have two methods to do this `mapResourceToDefaultScope` conversion.... the other is in `TokenCacheItemMigrationAdapter` though the return types are slightly different and _that_ implementation doesn't call `.trim()` which seems like a good protective measure....

If you're willing to take combining these bits of code as an addendum to this PR it'd be a nice bit of "gold plating" but I won't request changes based on this alone

Reply:
Oh, just kidding, I didn't realize that was a _new_ method. In that case, let's try to share code if we can...

I think having a method to transform a resource into a single scope makes sense, maybe put a wrapper function around it to box the result into a `String[]` - feel free to move out of `TokenCacheMigrationAdapter` and maybe move into `StringUtil` or some other appropriate class

Reply:
I'd be inclined to put this kind of helper function on the MicrosoftOAuth2Strategy.... since it's AAD specific logic.